packages/dev/core/src/Behaviors/Meshes/fadeInOutBehavior.ts

```diff
@@ -58,7 +67,28 @@ export class FadeInOutBehavior implements Behavior<Mesh> {
      * @param value if the object should fade in or out (true to fade in)
      */
     public fadeIn(value: boolean) {
+        // Cancel any pending updates
+        this._detachObserver();
+
+        // If fading in and already visible or fading out and already not visible do nothing
+        if (this._ownerNode && ((value && this._ownerNode.visibility >= 1) || (!value && this._ownerNode.visibility <= 0))) {
+            return;
+        }
+
         this._hovered = value;
+        if (!this._hovered) {
+            // Make the delay the negative of fadeout delay so the hoverValue is kept above 1 until
+            // fadeOutDelay has elapsed
+            this.delay = -this.fadeOutDelay;

```

Comment:
could we use a private _delay instead which would either be delay or -delay ??? 

It is strange to have both value separated ??? or it might be just me :-)

@raananw @Popov72 what do you think ?

Reply:
makes sense to me. no need to have both `this.delay` and `this.fadeOutDelay`, but `this.delay` is a public member, so that would break back-compat

Reply:
Sorry - correcting my last statement - one is fade in, one is fade out, both are needed, but the naming is problematic because of back-compat. We should probably leave it as is.

Reply:
> both are needed,

That is what I am questioning cause the system EITHER fades in OR out never both and currently the delay with fade out is mostly a bug so I would consider it a bug fix or I am missing smthg ???

Reply:
Is there a bug with fade out? If I understand correctly, this PR just adds a different timeout when fading out (as opposed to the same fade in and fade out time).